pyro/infer/tracegraph_elbo.py

```diff
@@ -190,9 +190,6 @@ def _get_traces(self, model, guide, *args, **kwargs):
         """
 
         for i in range(self.num_particles):
-            if self.enum_discrete:
```

Comment:
Enumeration is now controlled by hints in the `site['infer']` dict. If an inference algorithm does not implement enumeration, it can safely ignore those hints an sample rather than enumerate.